Sources/SwiftIfConfig/IfConfigEvaluation.swift

```diff
@@ -433,20 +434,77 @@ func evaluateIfConfig(
 }
 
 extension IfConfigClauseSyntax {
-  /// Determine whether this condition is "syntaxErrorsAllowed".
-  func syntaxErrorsAllowed(
-    configuration: some BuildConfiguration
+  /// Fold the operators within an #if condition, turning sequence expressions
+  /// involving the various allowed operators (&&, ||, !) into well-structured
+  /// binary operators.
+  public static func foldOperators(
+    _ condition: some ExprSyntaxProtocol
+  ) -> (folded: ExprSyntax, diagnostics: [Diagnostic]) {
+    var foldingDiagnostics: [Diagnostic] = []
+    let foldedCondition = OperatorTable.logicalOperators.foldAll(condition) { error in
+      foldingDiagnostics.append(contentsOf: error.asDiagnostics(at: condition))
+    }.cast(ExprSyntax.self)
+    return (folded: foldedCondition, diagnostics: foldingDiagnostics)
+  }
+
+  /// Determine whether the given expression, when used as the condition in
+  /// an inactive `#if` clause, implies that syntax errors are permitted within
+  /// that region.
+  public static func syntaxErrorsAllowed(
```

Comment:
Similar here: Does this method have to be public? I think internal would suffice.

Reply:
https://github.com/swiftlang/swift-syntax/pull/2806 makes it internal